src/main/java/com/ticketbooking/service/ReservationService.java

```diff
@@ -0,0 +1,323 @@
+        try {
+            String url = "http://" +
+                    request.getServerName() + ":" + request.getServerPort() + request.getContextPath() +
+                    "/confirmReservation?reservationId=" + reservationId + "&language=pl";
+            mailService.sendMail(mail, getMessage("mail.subject.text", null), url, false);
+        } catch (MessagingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private boolean checkSingleSeat(ReservationDto reservationDto, Screening screening) {
+        Map<Integer, List<Long>> map = getSeatsForScreening(reservationDto, screening);
+
+        map.entrySet().stream().forEach(System.out::println);
+
+        return verificationSeatsAfterReservation(map);
+    }
+
+    private boolean verificationSeatsAfterReservation(Map<Integer,List<Long>> map) {
+
+        final boolean[] verify = { true };
+        map.entrySet().stream()
+            .forEach(k -> {
+                        verify[0] = verifyPlacesInRow(k.getValue(), verify[0]);
+                    });
+
+        return verify[0];
+    }
+
+    private boolean verifyPlacesInRow(List<Long> row, boolean verify) {
+
+        if (!verify) return verify;
+        for (int ii = 1; ii < row.size()-1; ii++) {
+            if ( (row.get(ii) != 0) && (row.get(ii-1) == 0) && (row.get(ii+1) == 0)) {
+                verify = false;
+            }
+        }
+
+        return verify;
+    }
+
+    private Map<Integer, List<Long>> getSeatsForScreening(ReservationDto reservationDto, Screening screening) {
+        List<Seat> allSeatList = seatRepo.findAllByRoomId(screening.getRoom().getId());
+
+        Map<Integer, List<Long>> map = allSeatList.stream().collect(
+            Collectors.groupingBy(Seat::getRowNr, Collectors.mapping(Seat::getId, Collectors.toList())));
+
+        setReservedSeats(ticketRepo.findAllReservedByScreeningId(screening.getId()), map);
+        setReservedSeats(setReservedSeatsForCurrentReservation(reservationDto.getTickets()), map);
+
+        return map;
+    }
+
+    private List<Long> setReservedSeatsForCurrentReservation(List<TicketDto> tickets) {
+
+        return tickets.stream()
+            .map(t -> t.getSeatId())
+            .collect(Collectors.toList());
+    }
+
+    private void setReservedSeats(List<Long> seatList, Map<Integer, List<Long>> map) {
+        final int[] idx = { 0 };
+        map.entrySet().stream()
+            .map(v -> {
+                idx[0] = 0;
+                return v;
+            })
+            .forEach(k -> k.getValue().stream()
+                .forEach(v -> {
+                    if (seatList.contains(v)) {
+                        k.getValue().set(idx[0], 0L);
+                    }
+                    idx[0]++;
+                }));
+    }
+
+    private BigDecimal saveTickets(ReservationDto reservationDto, Screening screening, Reservation reservation) {
+        final BigDecimal[] sum = {BigDecimal.ZERO};
```

Comment:
po co nam tablica?

Reply:
Bo sam obiekt tablicy ma być finalny - to jest wymagane przez javę, natomiast wartości jej elementów można zmieniać co sprawdza się przy tym zastosowaniu.